packages/node/src/indexer/dictionary/v2/ethDictionaryV2.ts

```
@@ -0,0 +1,293 @@
+  EthDictionaryV2QueryEntry
+> {
+  protected buildDictionaryQueryEntries(
+    dataSources: SubqlDatasource[],
+  ): DictionaryV2QueryEntry {
+    const filteredDs = ethFilterDs(dataSources);
+    return buildDictionaryV2QueryEntry(filteredDs);
+  }
+
+  constructor(
+    endpoint: string,
+    nodeConfig: NodeConfig,
+    eventEmitter: EventEmitter2,
+    project: SubqueryProject,
+    chainId?: string,
+  ) {
+    super(
+      endpoint,
+      chainId ?? project.network.chainId,
+      nodeConfig,
+      eventEmitter,
+    );
+  }
+
+  /**
+   *
+   * @param startBlock
+   * @param queryEndBlock this block number will limit the max query range, increase dictionary query speed
+   * @param batchSize
+   * @param conditions
+   */
+  async getData(
+    startBlock: number,
+    queryEndBlock: number,
+    limit = MIN_FAT_FETCH_LIMIT,
+  ): Promise<DictionaryResponse<EthereumBlock & IBlockUtil> | undefined> {
+    const queryDetails = this.queriesMap?.getDetails(startBlock);
+    const conditions = queryDetails?.value;
+    queryEndBlock = this.metadata.end;
+
+    if (!conditions) {
+      return undefined;
+    }
+    const requestData = {
+      jsonrpc: '2.0',
+      method: FAT_BLOCKS_QUERY_METHOD,
+      id: 1,
+      params: [
+        startBlock,
+        queryEndBlock,
+        limit,
+        conditions,
+        { blockHeader: true, logs: true, transactions: { data: true } },
+      ],
+    };
+
+    try {
+      const response = await this.dictionaryApi.post(
+        this.dictionaryEndpoint,
+        requestData,
+        {
+          headers: {
+            'Content-Type': 'application/json',
+          },
+        },
+      );
+      const ethBlocks = this.convertResponseBlocks(response.data.result);
+      return {
+        batchBlocks: ethBlocks.blocks,
+        lastBufferedHeight: ethBlocks.end,
+      };
+    } catch (error) {
+      // Handle the error as needed
+      throw new Error(`Fat dictionary get capacity failed ${error}`);
```

This error is getting swallowed and the dictionary keeps making the same query